File: Python/ps_baekjoon/14676/main.py
import sys

readline, write = (sys.stdin.readline, sys.stdout.write)

n, m, k = [int(i) for i in readline().split()]
degree = [0] * (n + 1)
ea = [0] * (n + 1)
graph = [[] for _ in range(n + 1)]
for _ in range(m):
    frm, to = [int(i) for i in readline().split()]
    graph[frm].append(to)
    degree[to] += 1
for _ in range(k):
    opt, num = [int(i) for i in readline().split()]
    if opt == 1:
        if degree[num] == 0:
            if ea[num] == 0:
                for e in graph[num]:
                    degree[e] -= 1
            ea[num] += 1
        else:
            write("Lier!")
            exit(0)
    elif opt == 2:
        ea[num] -= 1
        if ea[num] < 0:
            write("Lier!")
            exit(0)
        if ea[num] == 0:
            for e in graph[num]:
                degree[e] += 1
write("King-God-Emperor")

File: C++/ps_baekjoon/14676/main.cpp
#include <bits/stdc++.h>
using namespace std;

constexpr int mxN = 1e5;
int n, m, k, degree[mxN + 1], ea[mxN + 1];
vector<vector<int>> graph;

int main()
{
    cin.tie(0)->sync_with_stdio(0);
    cin >> n >> m >> k;
    graph.resize(n + 1);
    while (m--)
    {
        int from, to;
        cin >> from >> to;
        ++degree[to];
        graph[from].push_back(to);
    }
    while (k--)
    {
        int opt, num;
        cin >> opt >> num;
        switch (opt)
        {
        case 1:
            if (degree[num] == 0)
            {
                if (ea[num]++ == 0)
                    for (const auto& e : graph[num])
                        --degree[e];
            }
            else
            {
                cout << "Lier!";
                exit(0);
            }
            break;
        case 2:
            if (--ea[num] < 0)
            {
                cout << "Lier!";
                exit(0);
            }
            if (ea[num] == 0)
                for (const auto& e : graph[num])
                    ++degree[e];
            break;
        default:
            break;
        }
    }
    cout << "King-God-Emperor";
    return 0;
}